### PR TITLE
Device CRD: remove configmap when no device is bound with

### LINKED ISSUE #478: configmap should be cleaned when no device is bound with
**What happened**:
Currently, the configmap will be kept even there is no device bound with, this will cause many useless leftover configmap, which has an empty profile with it.

**What you expected to happen**:
Those configmap mentioned above should be clean up.

**How to reproduce it (as minimally and precisely as possible)**:
- create a device
- delete a device
- list the configmap and check the content

**Anything else we need to know?**:

**Environment**:
- KubeEdge version: latest
- Hardware configuration: ALL
- OS (e.g. from /etc/os-release): UBUTNU16.04
- Kernel (e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by chendave**
Fix #478
Signed-off-by: Dave Chen <dave.chen@arm.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #478

### CODE REVIEW

**Comment by m1093782566** [comment]
cc @rohitsardesai83

File: cloud/pkg/devicecontroller/controller/downstream.go [hunk 1]
```diff
@@ -586,6 +586,15 @@ func (dc *DownstreamController) deleteFromConfigMap(device *v1alpha1.Device) {
 		dc.deleteFromDeviceProfile(device, nodeConfigMap)
 		// store new config map
 		dc.configMapManager.ConfigMap.Store(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0], nodeConfigMap)
+
+		// no device is bound to the configMap, then remove the configMap directly.
```
Comment (sids-b) [inline cloud/pkg/devicecontroller/controller/downstream.go]: Can we move line line 588 below this if to reduce 1 extra store operation before delete ?

**Comment by sids-b** [comment]
@chendave , would you like to join KubeEdge as a collaborator ?

File: cloud/pkg/devicecontroller/controller/downstream.go (hunk 1 above)
Comment (chendave) [reply cloud/pkg/devicecontroller/controller/downstream.go]: sure, will do later.

**Comment by chendave** [comment]
@sids-b , that would be great!  it's my honor!

**Comment by sids-b** [comment]
> @sids-b , that would be great! it's my honor!

Thanks for your contributions. Looking forward for active contributions :) 
@kevin-wangzefeng will send you invite soon

**Pull request merged**